.github/workflows/test-jenkins.yml

```
@@ -0,0 +1,231 @@
+
+            elif [ ${{ matrix.dir }} == "03_maven_tutorial" ]; then
+              ARG=maven
+              echo "Running $ARG tutorial"
+            else
+              # If directory is not one of the four directories, add it to temporary file
+              echo "internal examples have not been modified" 
+            fi
+            
+            
+      - name: checking files running the jenkins_init.sh with argument
+        run: |
+            if [ $(git diff --name-only HEAD^ HEAD | uniq | grep -c "${{ matrix.dir }}") -ne 0 ]; then
+              echo "Changed directories are $(git diff --name-only HEAD^ HEAD | uniq)"
+              ./jenkins_init.sh $ARG
+            else
+              echo "No Internal examples were modified"
+              echo 'NO_CHANGES=true' >> $GITHUB_ENV
+            fi
+
+      - name: Waiting for services to be ready
+        if: env.NO_CHANGES != 'true'
+        run: |
+          # After the curl request, the output is piped to the awk command. It is used to search for the message
+          # "Please wait while Jenkins is getting ready to work" in the curl output.
+          # If the message is found, awk exits with a non-zero status (1), and the loop continues.
+          # If the message is not found, the loop exits, and the "Jenkins is running" message is displayed.
+          timeout 60 bash -c 'until curl -s -f http://127.0.0.1:8080/login > /dev/null; do sleep 5; done' && echo "Jenkins is running" || echo "Jenkins is not running"
+          echo "Jenkins is ready"
+          JENKINS_VERSION=$(curl -s -I -k http://admin:admin@127.0.0.1:8080 | grep -i '^X-Jenkins:' | awk '{print $2}')
+          echo "Jenkins version is: $JENKINS_VERSION"
+          
+
+
+      - name: Run curl command to test the stack
+        if: env.NO_CHANGES != 'true'
+        run: |         
+          set -x
+          # Installing dependencies 
+          # To check Sthe version of Jenkins, load the top page or any .../api/* page and check for the X-Jenkins response header. This contains the version number of Jenkins, like "1.404" This is also a good way to check if an URL is a Jenkins URL.
+          JENKINS_VERSION=$(curl -s -I -k http://admin:admin@127.0.0.1:8080 | grep -i '^X-Jenkins:' | awk '{print $2}')
+          echo "Jenkins version is: $JENKINS_VERSION"
+          # Before launching a job, we need to create a token for the admin user
+          CRUMB=$(curl -s -k http://admin:admin@127.0.0.1:8080/crumbIssuer/api/xml?xpath=concat\(//crumbRequestField,%22:%22,//crumb\) -c cookies.txt)
+          echo "CRUMB was found."
+          TOKEN=$(curl -s -k 'http://admin:admin@127.0.0.1:8080/user/admin/descriptorByName/jenkins.security.ApiTokenProperty/generateNewToken' --data 'newTokenName=kb-token' -b cookies.txt -H $CRUMB | jq -r '.data.tokenValue')
+          echo "TOKEN was found."
+          # Let's launch a job. The token has been generated in the Jenkins UI (admin, settings, and so on)
```

```suggestion
          # Let's launch a job. The token has been generated in the previous steps.
```

My bad. I wrote this comment before finding how to do it with the command line. 😊